src/main/java/pl/edu/agh/iet/katabank/bankproduct/Deposit.java

```
@@ -0,0 +1,84 @@
+    private final int durationInMonths;
+    private boolean open;
+
+    public Deposit(Account account, BigDecimal initialBalance) {
+        this(account, initialBalance, LocalDate.now(), DEFAULT_DURATION);
+    }
+
+    public Deposit(Account account, BigDecimal initialBalance, LocalDate openDate, int durationInMonths) {
+        try {
+            account.withdraw(initialBalance);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(INCORRECT_AMOUNT_MESSAGE + initialBalance, e);
+        }
+
+        this.balance = initialBalance;
+        this.connectedAccount = account;
+        this.id = UUID.randomUUID();
+        this.openDate = openDate;
+        this.durationInMonths = durationInMonths;
+        this.open = true;
+    }
+
+    public Account getConnectedAccount() {
+        return this.connectedAccount;
+    }
+
+    public BigDecimal getBalance() {
+        return this.balance;
+    }
+
+    public LocalDate getOpenDate() {
+        return this.openDate;
+    }
+
+    public boolean isOpen() {
+        return this.open;
+    }
+
+    @Override
+    public Customer getOwner() {
+        return this.connectedAccount.getOwner();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Deposit deposit = (Deposit) o;
+        return Objects.equals(id, deposit.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    public void closeDeposit(LocalDate date) {
+        if (date == null || date.isBefore(this.openDate.plusMonths(durationInMonths))) {
+            throw new RuntimeException("Cannot close deposit on date: " + date);
+        }
+        BigDecimal closeBalance = this.balance;
+        this.balance = BigDecimal.ZERO;
+        this.connectedAccount.deposit(closeBalance);
+        this.open = false;
+    }
+}
```

Hi.
The closeDeposit method does not check if the deposit in question is open ore has already been closed. 
Possibly we should throw an exception if it has been.
You could add that, and a test for it.
Not a big deal from the "balance" point of view - as the balance of a closed deposit is ZERO anyway.